mesonbuild/compilers/compilers.py

```diff
@@ -753,7 +753,27 @@ def log_var(var, val):
         return preproc_flags, compile_flags, link_flags
 
     def get_options(self):
-        return {} # build afresh every time
+        opts = {} # build afresh every time
+
+        # Take default values from env variables.
```

Comment:
How does this change reconfiguration? Earlier we would always rebuild this from scratch, but now it's merged with compiler options and those are restored from coredata.

Reply:
Also, does this mean that `CFLAGS` from the environment are also added for the cross-compiler now? That must not happen.

Reply:
> How does this change reconfiguration? Earlier we would always rebuild this from scratch, but now it's merged with compiler options and those are restored from coredata.

I'm not sure to understand what you mean. When you run `ninja reconfigure` it pulls new flags from env? I don't think that was working like that previously, since flags were stored in coredata at first configuration. 

Before my patch the code was doing:
```
            if lang not in self.coredata.external_args:
                self.coredata.external_args[lang] = compile_args
```
So on reconfigure, lang is already in `self.coredata.external_args` and it won't set the value from env.

> Also, does this mean that CFLAGS from the environment are also added for the cross-compiler now? That must not happen.

No, interpreter.py, detect_compilers(), it calls get_options() only on the native compiler. It is the backend who decide if it takes those values from coredata, see for example ninja backend does:

```
            if not target.is_cross:
                # Link args added from the env: LDFLAGS. We want these to
                # override all the defaults but not the per-target link args.
                commands += self.environment.coredata.external_link_args(linker.get_language())
```